Orange/widgets/tests/test_settings_handler.py

```diff
@@ -64,6 +67,42 @@ def test_write_defaults(self):
 
         os.remove(settings_file)
 
+    def test_write_PermissionError(self):
+        """
+        Error may occur while trying to open file. Therefore file
+        cannot be closed. Tests if error is handled.
+        GH-2147
+        """
+        _, settings_file = mkstemp(suffix='.ini')
+        handler = SettingsHandler()
+        handler._get_settings_filename = lambda: settings_file
+        patch_target = "Orange.widgets.settings.open" if version_info >= (3, 5) \
+            else "builtins.open"
+        with patch(patch_target, side_effect=PermissionError):
+            handler.write_defaults()
+        try:
```

Comment:
This is not a way to solve the problem. If remove sometimes fails, figure out why and solve the cause.